File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(pot, loc) {
    let order = 0;
    pot = pot.map((val, idx) => ({val, idx}))
    while(pot.length > 0) {
        let maxEl = Math.max(...pot.map(el => el.val))
        let tmp = pot.shift()
        if(tmp.val === maxEl) {
            order++;
            if(tmp.idx === loc) return order;
        } else {
            pot.push(tmp)
        }
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    for i,priority in enumerate(priorities): priorities[i] = (i,priority)
    cnt = 0
    while priorities:
        tmp = priorities.pop(0)
        if not priorities or tmp[1] >= max(row[1] for row in priorities): 
            cnt += 1
            if tmp[0] == location: return cnt
        else: priorities.append(tmp)
    return